sycl/test-e2e/Matrix/joint_matrix_hip_mfma.hpp

```diff
@@ -0,0 +1,117 @@
+
+#include <sycl/sycl.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <random>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using sycl::ext::oneapi::bfloat16;
+
+template <typename, size_t M, size_t N> struct input_limit {
+  static constexpr int value = M * N;
+};
+
+template <> struct input_limit<int8_t, 16, 16> {
+  static constexpr auto value = 128;
+};
+
+template <> struct input_limit<int8_t, 32, 32> {
+  static constexpr auto value = 128;
+};
+
+template <typename InType, typename OutType, size_t M, size_t N, size_t K,
+          layout OutLayout>
+void hip_matrix_mfma() {
+  InType A[M * K];
+  InType B[K * N];
+  OutType C[M * N];
+  OutType D[M * N];
+  OutType E[M * N];
+
+  std::mt19937 gen(0);
+  std::uniform_real_distribution<float> dist(-100, 100);
+
+  for (auto i = 0; i < M * K; ++i) {
+    A[i] = static_cast<InType>(dist(gen));
+  }
+
+  for (auto i = 0; i < K * N; ++i) {
+    B[i] = static_cast<InType>(dist(gen));
+  }
+
+  for (auto i = 0; i < M * N; ++i) {
+    D[i] = 0;
+    C[i] = static_cast<OutType>(dist(gen));
+    if (OutLayout == layout::row_major)
+      E[i] = C[i];
+    else
+      E[(i % N) * M + int(i / M)] = C[i];
+  }
+
+  try {
+    auto defaultQueue = sycl::queue{};
+
+    auto bufA = sycl::buffer{A, sycl::range{M * K}};
+    auto bufB = sycl::buffer{B, sycl::range{K * N}};
+    auto bufC = sycl::buffer{C, sycl::range{M * N}};
+    auto bufD = sycl::buffer{D, sycl::range{M * N}};
+
+    defaultQueue
+        .submit([&](sycl::handler &cgh) {
+          sycl::accessor accA{bufA, cgh, sycl::read_only};
+          sycl::accessor accB{bufB, cgh, sycl::read_only};
+          sycl::accessor accC{bufC, cgh, sycl::read_only};
+          sycl::accessor accD{bufD, cgh, sycl::write_only};
+
+          cgh.parallel_for(
+              sycl::nd_range<2>{{4, 16}, {4, 16}}, [=](sycl::nd_item<2> idx) {
+                auto sg = idx.get_sub_group();
+                joint_matrix<sub_group, OutType, use::accumulator, M, N> sub_c;
+                joint_matrix<sub_group, InType, use::b, K, N, layout::row_major>
+                    sub_b;
+                joint_matrix<sub_group, InType, use::a, M, K, layout::col_major>
+                    sub_a;
+
+                joint_matrix_load(
+                    sg, sub_a,
+                    accA.template get_multi_ptr<access::decorated::yes>(), K);
```

Comment:
Here, the pointer here represents the starting address of the load so it should include the offset of the load as well. The offset calculation should not be done in the body of load_*_hip

Reply:
Also, can we use a more realistic example where M, N, K sizes in memory are bigger than M,N,K sizes of joint_matrix?

Reply:
Thanks @dkhaldi 
sure, I'll add more tests.